src/main/java/net/atos/zac/flowable/BPMNService.java

```diff
@@ -0,0 +1,101 @@
+package net.atos.zac.flowable;
+
+import static net.atos.zac.flowable.ZaakVariabelenService.VAR_ZAAKTYPE_OMSCHRIJVING;
+import static net.atos.zac.flowable.ZaakVariabelenService.VAR_ZAAKTYPE_UUUID;
+import static net.atos.zac.flowable.ZaakVariabelenService.VAR_ZAAK_IDENTIFICATIE;
+import static net.atos.zac.flowable.ZaakVariabelenService.VAR_ZAAK_UUID;
+
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.Map;
+import java.util.UUID;
+import java.util.logging.Logger;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+
+import org.flowable.common.engine.api.FlowableObjectNotFoundException;
+import org.flowable.engine.ProcessEngine;
+import org.flowable.engine.RepositoryService;
+import org.flowable.engine.RuntimeService;
+import org.flowable.engine.repository.ProcessDefinition;
+import org.flowable.engine.runtime.ProcessInstance;
+
+import net.atos.client.zgw.zrc.model.Zaak;
+import net.atos.client.zgw.ztc.model.Zaaktype;
+
+@ApplicationScoped
+@Transactional
+public class BPMNService {
+
+    private static final Logger LOG = Logger.getLogger(BPMNService.class.getName());
+
+    @Inject
+    private RepositoryService repositoryService;
+
+    @Inject
+    private RuntimeService runtimeService;
+
+    @Inject
+    private ProcessEngine processEngine;
+
+    public InputStream getProcessDiagram(final UUID zaakUUID) {
+        final var processInstance = findProcessInstance(zaakUUID);
+        final var processDefinition =
+                repositoryService.getProcessDefinition(processInstance.getProcessDefinitionId());
+        final var bpmnModel = repositoryService.getBpmnModel(processDefinition.getId());
+        final var processEngineConfiguration = processEngine.getProcessEngineConfiguration();
+        return processEngineConfiguration.getProcessDiagramGenerator()
+                .generateDiagram(bpmnModel, "gif",
```

Comment:
png

Reply:
Je kunt er png van maken maar dan krijgt het plaatje opeens een zwarte achtergrond en is redelijk onleesbaar.